.github/workflows/push-dev.yml

```diff
@@ -0,0 +1,79 @@
+name: Dev Changes
+
+on:
+  push:
+    branches:
+      - development
+
+jobs:
+  test:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+          cache: pip
+      - name: Install dependencies
+        run: make dev-env
+      - name: Run unit tests
+        run: make full-coverage-unit-tests
+      - name: Upload coverage report
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage_unit_tests_report
+          path: full_unit_tests.coverage
+
+  integration:
+    name: Integration tests
+    runs-on: [ self-hosted, Linux ]
+    container:
+      image: python:3.9.18
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install dependencies
+        run: make dev-env
+
+      - name: Run integration tests
+        env:
+          V3IO_API: ${{ secrets.V3IO_API }}
+          V3IO_ACCESS_KEY: ${{ secrets.V3IO_ACCESS_KEY }}
+          V3IO_FRAMESD: ${{ secrets.V3IO_FRAMESD }}
+        run: make coverage-integration
+
+      - name: Upload coverage report
```

Comment:
please add a comment explaining that the file is not really "uploaded" - it's saved locally (in the runner?)
is there a need to clean this files at the end of run?

Reply:
you were right, there was a danger of overriding artifacts, that's why I switch to use artifact name (id) with combination of github.run_id  in order to get a fully unique identifier.

Moreover, I added a retention date for each artifact of 30 days, In order to avoid bulking up too many reports.
90 days is the default if we are not mentioning something else in the yml file.

Reply:
Another useful feature:
you can download the coverage report directly from the run:

![image](https://github.com/user-attachments/assets/eb5948c9-e4b3-445b-8fc1-c74b84352368)